### ISSUE TITLE
core package test refactor

### ISSUE THREAD

**Issue by Tlacenka** [code-pushup/cli#215]
As a contributor to Code PushUp, I would like for the tests to be consistently structured and follow good practices.
There are several areas in which I identified room for improvement. In this issue, the following targeted changes should be made.

## Coverage
- [x] Add a happy path test to `read-code-pushup-config` and invalid config error scenario.
- [x] Test both the regular approach and ESM plugin approach once #248 is merged.

In both `collect` and `collect-and-persist`, the main cases should be tested. This means:
- [x] Add a test case for no plugins passed to `collect`.
- [x] Add a test case for explicitly adding multiple output options for `collect-and-persist` and explicitly mark each use case.

## Files
These tests are mostly in scope of **integration tests** (except for `json-to-gql`). Hence ,they should be categorised as integration tests and structured accordingly.
**Note**: `nx` configuration of integration vs unit tests is OOS. This is only about the location and naming of the files.
- [x] Each test file name reflects that it is an integration test (to be finalised).
- [x] Move all configurations and mocks for FS and other under `testing-utils`.
- [x] Remove the unused `types.ts` file.
- [x] It should be decided whether `read-code-pushup-config` and `json-to-gql` and the progress configs belong to this package - in my opinion, the first two functions belong in `utils` and the configurations should be moved to `testing-utils`. However, perhaps there is a reason to keep them in `core`?
- [x] Keep only one `README` file. Currently, there are two - one in the root folder and one under `docs`.
- [x] Merge `DEFAULT_TESTING_CLI_OPTIONS` from `constant.ts`, also used in other packages, under `testing-utils` or use hardcoded values for better readability where applicable.

## Good practices
In order to create and maintain tests easily, their readability should be prioritised. This means minimising boiler plate code, keeping the test flow linear and naming things based on their purpose.
- [x] Instead of mocking console, use `spyOn` for `console.log` in combination with `shouldHaveBeenCalledWith()`.
- [x] Move `portal-client`, `fs` and `fs/promises` mock setup to `testing-utils` and share setup for all FS vs upload flows there.
- [x] Remove boiler plate regarding `pluginConfig` in `execute-plugin`. Having one valid configuration and then overriding its options without the need for helper functions would increase the readability of the tests.
- [x] Remove branching logic in `collect-and-persist` and explicitly state which formats are tested in which file.
- [x] Improve test descriptions and assertions purpose in `persist` - it is not clear when JSON vs `stdout` vs `MD` is created.
- [x] Fix `not.toThrow('no such file...')` in `persist` to either check thrown error message or to check the result did not throw (and did not produce any message).
- [x] Reduce usage of `reportPath` and replace it with a hardcoded value - all tests expect the same path and concrete file format(s). If we want to test different paths, let's create a separate isolated test for that.
- [x] Improve naming of functions `report()`, `auditReport()` and other to signify they are mocks, create only one version and move that to `testing-utils` to be reused.
- [x] Replace `date` field validation in `execute-plugin` with a custom `toBeISODate()` assertion.
- [x] Mock `fs` in `read-code-pushup-config` tests. While `bundleRequire` should not be mocked, it doesn't matter where the file it parses lives. right now, unnecessary one-line files were added.

Note: Remaining coverage related tasks were moved to a separate issue #356
